### PR TITLE
Fix for CLOUD-3560, Define 7.3.1 and XP 1.0 env

### PR DESCRIPTION
**Pull request by jfdenise**
Issue: https://issues.redhat.com/browse/CLOUD-3560

### CODE REVIEW

**Review by luck3y — APPROVED** [review]

File: modules/eap-73-env/7.3.1/module.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      value: "/opt/eap/standalone/deployments"
+    - name: "com.redhat.dev-mode"
+      value: "DEBUG:true"
+      description: "Environment variable used to enable development mode (debugging). A value of true will enable development mode."
+    - name: "com.redhat.dev-mode.port"
+      value: "DEBUG_PORT:8787"
+      description: "Environment variable used to specify the debug port. If not set, the default EAP debug port will be used (8787). Only applicable when development mode is enabled."
+envs:
+    - name: "WILDFLY_VERSION"
+      value: "7.3.0.GA-redhat-SNAPSHOT"
```
Comment (jmesnil) [inline modules/eap-73-env/7.3.1/module.yaml:24]: must be updated to 7.3.1.GA-redhat-00001 (when the tag is known)

File: modules/eap-xp-10-env/1.0.0/module.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+schema_version: 1
+
+name: "eap-xp-10-env"
+description: "JBoss Enterprise Application Platform XP 1.0 env and labels"
```
Comment (jmesnil) [inline modules/eap-xp-10-env/1.0.0/module.yaml:4]: "JBoss Enterprise Application Platform expansion pack 1.0 env and labels"

File: modules/eap-xp-10-env/1.0.0/module.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      value: "/opt/eap/standalone/deployments"
+    - name: "com.redhat.dev-mode"
+      value: "DEBUG:true"
+      description: "Environment variable used to enable development mode (debugging). A value of true will enable development mode."
+    - name: "com.redhat.dev-mode.port"
+      value: "DEBUG_PORT:8787"
+      description: "Environment variable used to specify the debug port. If not set, the default EAP debug port will be used (8787). Only applicable when development mode is enabled."
+envs:
+    - name: "WILDFLY_VERSION"
+      value: "7.3.0.GA-redhat-SNAPSHOT"
```
Comment (jmesnil) [inline modules/eap-xp-10-env/1.0.0/module.yaml:24]: must be updated to 7.3.1.GA-redhat-00001 (when the tag is known)

File: modules/eap-xp-10-env/1.0.0/module.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+schema_version: 1
+
+name: "eap-xp-10-env"
+description: "JBoss Enterprise Application Platform XP 1.0 env and labels"
+version: "1.0.0"
+
+labels:
+    - name: "org.jboss.product"
+      value: "eap-xp"
+    - name: "org.jboss.product.version"
```
Comment (jmesnil) [inline modules/eap-xp-10-env/1.0.0/module.yaml:10]: @kabir  out of curiosity, is the XP manager setting these 2 properties (`org.jboss.product` and `org.jboss.product.version`) as System properties or environment variables?

File: modules/eap-xp-10-env/1.0.0/module.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      value: "1.0"
```
Comment (jmesnil) [inline modules/eap-xp-10-env/1.0.0/module.yaml:11]: regarding the value of the XP version, I think we should keep it with a x.y.z format similar to the EAP version(so it should be `1.0.0.`).

@kabir Could you confirm the XP version format please?

File: modules/eap-xp-10-env/1.0.0/module.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    - name: "WILDFLY_VERSION"
+      value: "7.3.0.GA-redhat-SNAPSHOT"
+    - name: "LAUNCH_JBOSS_IN_BACKGROUND"
+      value: "true"
+    - name: "JBOSS_PRODUCT"
+      value: "eap-xp"
+    - name: "JBOSS_EAP_VERSION"
+      value: "7.3.1"
+    - name: "PRODUCT_VERSION"
+      value: "1.0"
```
Comment (jmesnil) [inline modules/eap-xp-10-env/1.0.0/module.yaml:32]: same remark as above (`1.0.0`)

File: modules/eap-xp-10-env/1.0.0/module.yaml (hunk 2 above)
Comment (jfdenise) [reply modules/eap-xp-10-env/1.0.0/module.yaml:4]: Resolved in https://github.com/jboss-container-images/jboss-eap-7-image/pull/185

File: modules/eap-xp-10-env/1.0.0/module.yaml (hunk 5 above)
Comment (jfdenise) [reply modules/eap-xp-10-env/1.0.0/module.yaml:11]: Resolved in https://github.com/jboss-container-images/jboss-eap-7-image/pull/185

File: modules/eap-xp-10-env/1.0.0/module.yaml (hunk 6 above)
Comment (jfdenise) [reply modules/eap-xp-10-env/1.0.0/module.yaml:32]: Resolved in https://github.com/jboss-container-images/jboss-eap-7-image/pull/185

**Pull request merged**